ic-agent/src/agent/mod.rs

```diff
@@ -260,33 +265,20 @@ impl Agent {
         Ok(Agent {
             nonce_factory: config.nonce_factory,
             identity: config.identity,
-            ingress_expiry_duration: config
-                .ingress_expiry_duration
+            ingress_expiry: config
+                .ingress_expiry
                 .unwrap_or_else(|| Duration::from_secs(300)),
-            root_key: Arc::new(RwLock::new(Some(IC_ROOT_KEY.to_vec()))),
+            root_key: Arc::new(RwLock::new(IC_ROOT_KEY.to_vec())),
             transport: config
                 .transport
                 .ok_or_else(AgentError::MissingReplicaTransport)?,
+            backoff_initial_interval: config.backoff_initial_interval,
+            backoff_max_elapsed_time: config.backoff_max_elapsed_time,
+            backoff_max_interval: config.backoff_max_elapsed_time,
+            backoff_multiplier: config.backoff_multiplier,
         })
     }
 
-    /// Set the transport of the [`Agent`].
-    pub fn set_transport<F: 'static + Transport>(&mut self, transport: F) {
-        self.transport = Arc::new(transport);
-    }
-
-    /// Set the identity provider for signing messages.
-    ///
-    /// NOTE: if you change the identity while having update calls in
-    /// flight, you will not be able to [Agent::poll] the status of these
-    /// messages.
-    pub fn set_identity<I>(&mut self, identity: I)
-    where
-        I: 'static + Identity,
-    {
-        self.identity = Arc::new(identity);
-    }
-
```

Comment:
unintentional deletion? set_identity is here on purpose.

Reply:
Actually it was intentional. Do we really need to change the identity and transport of an already existing agent ? Especially given the working message in the comments ?

Reply:
Yes. For the identity, a developer does not necessarily want to use the same identity for every call (and `poll` is little-used); for the transport, it is used in `dfx canister sign` specifically to wrap an existing agent.

Reply:
ok, I reverted it for now. I can imagine that a user would like to set a new identity for different requests. However, setting a different transport may be a bit off. I would imagine if someone needs a new transport they would just create a new agent.